provider/virtio_fs_utils.py

```diff
@@ -0,0 +1,209 @@
+    """
+    if params.get("os_type", "windows") == "windows":
+        error_context.context("Running viofs basic io test via psexec",
+                              LOG_JOB.info)
+        cmd_dd_win = params.get("virtio_fs_cmd_dd_win",
+                                "dd if=/dev/random of=%s bs=1M count=100")
+        test_file = params.get('virtio_fs_test_file', "virtio_fs_test_file")
+        io_timeout = params.get("fs_io_timeout", 120)
+        fs_source = params.get("fs_source_dir", "virtio_fs_test/")
+        fs_target = params.get("fs_target", "myfs")
+        base_dir = params.get("fs_source_base_dir",
+                              data_dir.get_data_dir())
+        if not os.path.isabs(fs_source):
+            fs_source = os.path.join(base_dir, fs_source)
+        host_data = os.path.join(fs_source, test_file)
+
+        session = vm.wait_for_login()
+        driver_letter = get_virtiofs_driver_letter(test, fs_target, session)
+        fs_dest = "%s:" % driver_letter
+        guest_file = os.path.join(fs_dest, test_file)
+        cmd_io_test = "%systemdrive%\\cmd_io_test.bat"
+
+        error_context.context("Creating the test file(cmd_io_test.bat) "
+                              "on guest", LOG_JOB.info)
+        session.cmd("echo " + cmd_dd_win % guest_file + " > " + cmd_io_test,
+                    io_timeout)
+
+        psexec_path = install_psexec(vm)
+        try:
+            error_context.context("Execute the cmd_io_test.bat on guest",
+                                  LOG_JOB.info)
+            dns = params.get("dns", "")
+            dns += "\\" if dns else ""
+            session.cmd(psexec_path + " /accepteula -u " + dns + usernm +
+                        " -p " + pwd + " " + cmd_io_test)
+
+            guest_file_win = guest_file.replace("/", "\\")
+            cmd_md5 = params.get("cmd_md5", '%s: && md5sum.exe %s')
+            cmd_md5_vm = cmd_md5 % (driver_letter, guest_file_win)
+            md5_guest = session.cmd_output(cmd_md5_vm,
+                                           io_timeout).strip().split()[0]
+            error_context.context("md5 of the guest file: %s" % md5_guest,
+                                  LOG_JOB.info)
+            md5_host = process.run("md5sum %s" % host_data,
+                                   io_timeout).stdout_text.strip().split()[0]
+            error_context.context("md5 of the host file: %s" % md5_host,
+                                  LOG_JOB.info)
+            if md5_guest != md5_host:
+                test.fail('The md5 value of host is not same to guest.')
+            else:
+                error_context.context("The md5 of host is as same as md5 of "
+                                      "guest.", LOG_JOB.info)
+        finally:
+            error_context.context("Delete the test file from host.",
+                                  LOG_JOB.info)
+            os.remove(host_data)
+
+        create_sub_folder_test(params, session, fs_dest, fs_source)
```

Comment:
this function also need to be executed by different user, now ad_user not does it.